src/python.cc

```diff
@@ -355,6 +356,12 @@ ModelInstanceState::~ModelInstanceState()
   }
 }
 
+ModelState*
```

Comment:
You should return a shared pointer. Is ModelInstance supposed to be a struct or class? If it is a struct, just make model_state a public variable. If not, then don't return raw data. Rather, create methods that can be applied in the object.

Reply:
For example, you really only need `name` and `max_batch_size`, so create two methods that return just that data, so you are not passing pointers to internal data. That is bad and can be prone to problems down the road.  For example...

```
ModelState * state = instance_state->GetModelState();

.... Do useful work ....

delete state;  // oops, you just hosed the system by accident.
```

Reply:
Even more importantly, if you do return a pointer, make it a const pointer unless you intend to alter the data.

Reply:
> You should return a shared pointer. Is ModelInstance supposed to be a struct or class? If it is a struct, just make model_state a public variable. If not, then don't return raw data. Rather, create methods that can be applied in the object.

ModelInstanceState is a class. 

I see, thanks for the comment. I ended up deleting that method. It looks like that field was available in the parent class.